File: UINavigationExtensionDemo/Feature/Basic/Controllers/ViewController12_Modal.m
//
//  ViewController12_Modal.m
//  UINavigationExtensionDemo
//
//  Created by Leo Lee on 2020/10/26.
//

#import <UINavigationExtension/UINavigationExtension.h>

#import "ViewController12_Modal.h"
#import "UIImage+NavigationBar.h"
#import "UIColor+RandomColor.h"

@interface ViewController12_Modal ()

@end

@implementation ViewController12_Modal

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIButton *closeButton = [UIButton buttonWithType:UIButtonTypeCustom];
    closeButton.frame = CGRectMake(0, 0, 36, 36);
    closeButton.backgroundColor = [UIColor clearColor];
    [closeButton setImage:[[UIImage imageNamed:@"NavigationBarClose"] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] forState:UIControlStateNormal];
    [closeButton setTintColor:UIColor.customTitleColor];
    [closeButton addTarget:self action:@selector(clickCloseButton:) forControlEvents:UIControlEventTouchUpInside];
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithCustomView:closeButton];
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    if (@available(iOS 13.0, *)) {
        return UIStatusBarStyleDarkContent;
    } else {
        return UIStatusBarStyleDefault;
    }
}

- (void)clickCloseButton:(UIButton *)button {
    [self dismissViewControllerAnimated:YES completion:NULL];
}

@end

File: Example/NXNavigationExtensionSwiftDemo/Basic/ViewController12_Modal.swift
//
//  ViewController12_Modal.swift
//  NXNavigationExtensionSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class ViewController12_Modal: BaseTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        let closeButton = UIButton(type: .custom)
        closeButton.frame = CGRect(x: 0, y: 0, width: 36.0, height: 36.0)
        closeButton.backgroundColor = .clear
        closeButton.setImage(UIImage(named: "NavigationBarClose")?.withRenderingMode(.alwaysTemplate), for: .normal)
        closeButton.tintColor = .customTitle
        closeButton.addTarget(self, action: #selector(clickCloseButton(_:)), for: .touchUpInside)
        navigationItem.leftBarButtonItem = UIBarButtonItem(customView: closeButton)
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        if #available(iOS 13.0, *) {
            return .darkContent
        } else {
            return .default
        }
    }
    
}

@objc
extension ViewController12_Modal {
    
    private func clickCloseButton(_ button: UIButton) {
        dismiss(animated: true, completion: nil)
    }
    
}
